packages/react-reconciler/src/ReactFiber.js

```diff
@@ -278,7 +278,7 @@ const createFiber = function(
   return new FiberNode(tag, pendingProps, key, mode);
 };
 
-function shouldConstruct(Component) {
+export function shouldConstruct(Component: Function) {
```

Comment:
It doesn't make sense that other files import this function from ReactFiber. It doesn't have anything to do with the Fiber.

However, the way it is used, is to upgrade the tag. Maybe we could instead export a function like `export function reassignTag(fiber: Fiber, Component: Function): TypeOfWork` that does the work of figuring out which tag should've been used.

Then in mountIndeterminateComponent you use the resulting tag to figure out which branch to take.

That way, the logic about picking tag is fully encapsulated in ReactFiber.